src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -360,30 +360,18 @@ def is_device(dev):
             return False
 
     # fallback to stat
-    return _stat_is_device(os.lstat(dev).st_mode)
+    return _stat_is_device(os.lstat(dev).st_mode) and not is_partition(dev)
 
 
-def is_partition(dev):
+def is_partition(dev: str) -> bool:
     """
     Boolean to determine if a given device is a partition, like /dev/sda1
     """
     if not os.path.exists(dev):
         return False
-    # use lsblk first, fall back to using stat
-    TYPE = lsblk(dev).get('TYPE')
-    if TYPE:
-        return TYPE == 'part'
 
-    # fallback to stat
-    stat_obj = os.stat(dev)
-    if _stat_is_device(stat_obj.st_mode):
-        return False
-
-    major = os.major(stat_obj.st_rdev)
-    minor = os.minor(stat_obj.st_rdev)
-    if os.path.exists('/sys/dev/block/%d:%d/partition' % (major, minor)):
-        return True
-    return False
+    partitions = get_partitions()
```

Comment:
backport of get_partitions seems to be missing.
```
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr Traceback (most recent call last):
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr   File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/prepare.py", line 240, in safe_prepare
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr     self.prepare()
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr   File "/usr/lib/python3.6/site-packages/ceph_volume/decorators.py", line 16, in is_root
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr     return func(*a, **kw)
2024-01-15T22:23:23.862 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr   File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/prepare.py", line 351, in prepare
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr     block_lv = self.prepare_data_device('block', osd_fsid)
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr   File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/prepare.py", line 196, in prepare_data_device
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr     if disk.is_partition(device) or disk.is_device(device):
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr   File "/usr/lib/python3.6/site-packages/ceph_volume/util/disk.py", line 373, in is_partition
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr     partitions = get_partitions()
2024-01-15T22:23:23.863 INFO:teuthology.orchestra.run.smithi017.stderr:/bin/podman: stderr NameError: name 'get_partitions' is not defined

```


https://pulpito.ceph.com/yuriw-2024-01-15_22:02:29-rados-wip-yuri2-testing-2024-01-12-1128-pacific-distro-default-smithi/

Reply:
Added 00ba00fdfab8cb4969d46838e44cff3a03fc16ca